viadot/sources/cloud_for_customers.py

```diff
@@ -0,0 +1,256 @@
+
+        Returns:
+            records (List[Dict[str, Any]]): The records extracted from url.
+        """
+        tmp_full_url = deepcopy(url)
+        tmp_filter_params = deepcopy(self.filter_params)
+        records = []
+        while url:
+            response = self.get_response(tmp_full_url, filter_params=tmp_filter_params)
+            response_json = response.json()
+            if isinstance(response_json["d"], dict):
+                # ODATA v2+ API
+                new_records = response_json["d"].get("results")
+                url = response_json["d"].get("__next", None)
+            else:
+                # ODATA v1
+                new_records = response_json["d"]
+                url = response_json.get("__next", None)
+
+            # prevents concatenation of previous url's with filter_params with the same filter_params
+            tmp_filter_params = None
+            tmp_full_url = url
+
+            if hasattr(new_records, "__iter__"):
+                records.extend(new_records)
+        return records
+
+    def extract_records(self) -> List[Dict[str, Any]]:
+        """Downloads records from url or report_url if present.
+
+        Returns:
+            records (List[Dict[str, Any]]): The records extracted from url.
+        """
+        if self.is_report:
+            return self._extract_records_from_report_url(url=self.report_url)
+        else:
+            return self._extract_records_from_url(url=self.full_url)
+
+    def get_entities(
+        self, dirty_json: Dict[str, Any], url: str
+    ) -> List[Dict[str, Any]]:
+        """Extracts entities from request.json().
+
+        Args:
+            dirty_json (Dict[str, Any]): request.json() dict from response to API.
+            url (str): The url to fetch metadata from.
+
+        Returns:
+            entities (List[Dict[str, Any]]): list filled with entities.
+        """
+
+        metadata_url = self.create_metadata_url(url)
+        column_maper_dict = self.get_mapping_property_to_sap_level(metadata_url)
+        entities = []
+        for element in dirty_json["d"]["results"]:
+            new_entity = {}
+            for key, object_of_interest in element.items():
+                if key not in ["__metadata", "Photo", "", "Picture"]:
+                    if "{" not in str(object_of_interest):
+                        new_key = column_maper_dict.get(key)
+                        if new_key:
+                            new_entity[new_key] = object_of_interest
+                        else:
+                            new_entity[key] = object_of_interest
+            entities.append(new_entity)
+        return entities
+
+    def get_mapping_property_to_sap_level(self, url: str = None) -> Dict[str, str]:
+        """Creates Dict mapping property Name to value of sap label.
+
+        Args:
+            url (str, optional): The url to fetch metadata from.
+
+        Returns:
+            Dict[str, str]: Property Name to value of sap label Dict.
```

Comment:
sap -> SAP; what's the "Dict" at the end?

Reply:
The sentence is basically: "x to y dictionary"
x being "Property Name"
y being "value of sap label"

Reply:
what is SAP label dict then?

Reply:
"SAP lebel" is not a dict, it is a str, "property value" also is a str.
the "Dict" is not referring to "SAP label" a dict, it is referring to the object to be returned, which is a "x to y Dict"
but maybe this is confusing, I will remove the "Dict"